### PR TITLE
Added a minimal logger so we know what's going on with iw4x updates p…

### PR DESCRIPTION
**Pull request by Rackover**
…ermanently stuck

### CODE REVIEW

File: src/common/utils/logger.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include "logger.hpp"
+
+const std::string utils::logger::LOG_FILE_NAME = "xlabs.log";
+std::ofstream utils::logger::log_file_stream;
+std::mutex utils::logger::lock;
```
Comment (diamante0018) [inline src/common/utils/logger.cpp:5]: Have you checked out the other class utils class named_mutex?
There are examples on how to use it in the repo.

File: src/common/utils/logger.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include "logger.hpp"
+
+const std::string utils::logger::LOG_FILE_NAME = "xlabs.log";
+std::ofstream utils::logger::log_file_stream;
+std::mutex utils::logger::lock;
+
+void utils::logger::write(std::string msg)
+{
+	lock.lock();
```
Comment (diamante0018) [inline src/common/utils/logger.cpp:9]: Not ideal, I feel this should be used in conjunction with named mutex class with the std::unique_lock wrapper so it does all the good RAII stuff

**Review by diamante0018 — CHANGES_REQUESTED** [review]
I like this change
Please be consinstent with your style of string concatenation
Sometimes it's like `"..."+val+"..."` other times there is a space between the `+` and the `"`

**Comment by diamante0018** [comment]
After some time spent googling  I decided to keep static inline in the header because it should make the compiler inline the calls to the function which is what we want if the std::source_location to be accurate

File: src/common/utils/logger.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		}
+
+		void write_to_log(const std::string& line)
+		{
+			std::unique_lock<std::mutex> _(logger_mutex);
+
+			try
+			{
+				auto& log_file_stream = get_stream();
+				log_file_stream << line << std::endl;
```
[the remark this answers is not in the document]
Comment (diamante0018) [reply src/common/utils/logger.cpp:28]: the try catch is for the exceptions that << overloaded function may (don't know which but I don't trust io functions) throw it's not to check if the stream is open or not. 
I will add a check to see if the stream is open.
If an exceptions is caught it a messagebox can be displayed i guess

**Review by momo5502 — APPROVED** [review]

**Pull request merged**